corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -331,13 +331,19 @@ def __init__(self, project, user, params):
 
     def validate_state(self):
         check_version(self.params.version)
-        if self.last_sync_log and self.params.state_hash:
-            parsed_hash = CaseStateHash.parse(self.params.state_hash)
-            computed_hash = self.last_sync_log.get_state_hash()
-            if computed_hash != parsed_hash:
-                raise BadStateException(expected=computed_hash,
-                                        actual=parsed_hash,
-                                        case_ids=self.last_sync_log.get_footprint_of_cases_on_phone())
+        if self.last_sync_log:
+            if type(self.last_sync_log) != self.sync_log_class:
```

Comment:
I think at the time of writing I forgot that `SimplifiedSyncLog` didn't extend `SyncLog`. will fix